### DISCUSSION TITLE
Is it possible to detect if a user typed out a command vs. a user used the abbreviation?

### DISCUSSION THREAD

**Discussion by mainrs** [olets/zsh-abbr#123] [Q&A] [answered]
Maybe through an internal variable that is set to 1 or contains the abbreviation that has been used?

**Comment by olets** [comment]
Interesting idea. No such variables are currently exported, but I think it would be straight forward.

What's the use case? So I can make sure I'm thinking about it correctly.

**Reply by mainrs** [reply]
Mostly this issue here: https://github.com/MichaelAquilina/zsh-you-should-use/issues/129#issuecomment-2115936184

The other zsh plugin gives you a reminder if you use a full command instead of an alias. Someone implemented interoperability with `zsh-abbr`. There is no way to detect if a command has been replaced or not. Thus, there can be no conditional reminder for an abbreviation. The current implementation will always remind the user of the abbreviation if it detects that the substituted command in the history. 

That's why I opened the discussion question. I think it would be great to have to variables. One that exposes the abbreviation that was used and one the substituted command. Maybe `ZSH_ABBR` and `ZSH_ABBR_SUBSTITUTE`? And if not set, then it means that no abbreviation has been used.

**Reply by olets** [reply]
Gotcha. A way to tell whether an abbreviation was expanded will help, and might be worth adding no matter what as something for others to hook into.

zsh-should-you-use integration is made complicated by multi-word global abbreviations, and even more so by the upcoming "regular abbreviation prefixes" feature (https://github.com/olets/zsh-abbr/discussions/114).

Considering this blocked by 114. Can revisit some time after that's released.

**Reply by olets** [reply]
After reading the proposed implementation, and thinking about this some more:

It might be reasonable to have the bulk of the work, not just a "was an abbreviation used" variable, live in zsh-abbr. Commented on the zsh-you-should-use issue.

**Comment by olets** [comment, marked as answer]
Belated update: support for reminders was added in [v6.0.0](https://github.com/olets/zsh-abbr/releases/tag/v6.0.0)!
